scipy/signal/_signaltools.py

```diff
@@ -3540,170 +3566,152 @@ def resample(x, num, t=None, axis=0, window=None, domain='time'):
 
     Notes
     -----
-    The argument `window` controls a Fourier-domain window that tapers
-    the Fourier spectrum before zero-padding to alleviate ringing in
-    the resampled values for sampled signals you didn't intend to be
-    interpreted as band-limited.
-
-    If `window` is a function, then it is called with a vector of inputs
-    indicating the frequency bins (i.e. fftfreq(x.shape[axis]) ).
-
-    If `window` is an array of the same length as `x.shape[axis]` it is
-    assumed to be the window to be applied directly in the Fourier
-    domain (with dc and low-frequency first).
-
-    For any other type of `window`, the function `scipy.signal.get_window`
-    is called to generate the window.
-
-    The first sample of the returned vector is the same as the first
-    sample of the input vector.  The spacing between samples is changed
-    from ``dx`` to ``dx * len(x) / num``.
-
-    If `t` is not None, then it is used solely to calculate the resampled
-    positions `resampled_t`
-
-    As noted, `resample` uses FFT transformations, which can be very 
-    slow if the number of input or output samples is large and prime; 
-    see :func:`~scipy.fft.fft`. In such cases, it can be faster to first downsample 
-    a signal of length ``n`` with :func:`~scipy.signal.resample_poly` by a factor of 
-    ``n//num`` before using `resample`. Note that this approach changes the 
-    characteristics of the antialiasing filter.
+    This function uses the more efficient one-sided FFT, i.e. `~scipy.fft.rfft` /
+    `~scipy.fft.irfft`, if `x` is real-valued and in the time domain.
+    Else, the two-sided FFT, i.e., `~scipy.fft.fft` / `~scipy.fft.ifft`, is used
+    (all FFT functions are taken from the `scipy.fft` module).
+
+    If a `window` is applied to a real-valued `x`, the one-sided spectral windowing
+    function is determined by taking the average of the negative and the positive
+    frequency component. This ensures that real-valued signals and complex signals with
+    zero imaginary part are treated identically. I.e., passing `x` or passing
+    ``x.astype(np.complex128)`` produce the same numeric result.
+
+    If the number of input  or output samples are prime or have few prime factors, this
+    function may be slow due to utilizing FFTs. Consult `~scipy.fft.prev_fast_len` and
+    `~scipy.fft.next_fast_len` for determining efficient signals lengths.
+    Alternatively, utilizing `resample_poly` to calculate an intermediate signal (as
+    illustrated in the example below) can result in significant speed increases.
+
+    `resample` is intended to be used for periodic signals with equidistant sampling
+    intervals. For non-periodic signals, `resample_poly` may be a better choice.
```

Comment:
Consider stressing this in the See Also link to `resample_poly`.

Reply:
Added in commit 76636f9e7.